schema/messaging.proto

```diff
@@ -0,0 +1,482 @@
+import "google/protobuf/timestamp.proto";
+import "google/rpc/error_details.proto";
+
+package google.showcase.v1alpha3;
+
+option go_package = "github.com/googleapis/gapic-showcase/server/genproto";
+option java_package = "com.google.showcase.v1alpha3";
+option java_multiple_files = true;
+
+option (google.api.metadata) = {
+  product_uri: "https://github.com/googleapis/gapic-showcase"
+};
+
+// This resource is defined in google/showcase/v1alpha3/user.proto. Since
+// google/showcase/v1alpha3/user.proto is not imported from this file, the
+// resource needs to be defined here to be used in this API.
+option (google.api.resource_definition) = {
+  name: "User",
+  path: "users/{user_id}"
+};
+
+// A resource set defining the possible parent paths of a	blurb.
+option (google.api.resource_set_definition) = {
+  name: "BlurbParent"
+  resources: [{
+    name: "Profile",
+    path: "users/{user_id}/profile"
+  }]
+  resource_references: [
+    "Room"
+  ]
+};
+
+// A simple messaging service that implements chat rooms and profile posts.
+//
+// This messaging service showcases the features that API clients
+// generated by gapic-generators implement.
+service Messaging {
+  // Creates a room.
+  rpc CreateRoom(CreateRoomRequest) returns (Room) {
+    option (google.api.http) = {
+      post: "/v1alpha3/rooms"
+      body: "*"
+    };
+    option (google.api.method_signature) = {
+      fields: ["room.display_name", "room.description"]
+    };
+  }
+
+  // Retrieves the Room with the given resource name.
+  rpc GetRoom(GetRoomRequest) returns (Room) {
+    option (google.api.http) = {
+      get: "/v1alpha3/{name=rooms/*}"
+    };
+    option (google.api.method_signature) = {
+      fields: ["name"]
+    };
+  }
+
+  // Updates a room.
+  rpc UpdateRoom(UpdateRoomRequest) returns (Room) {
+    option (google.api.http) = {
+      patch: "/v1alpha3/{room.name=rooms/*}"
+      body: "*"
+    };
+  }
+
+  // Deletes a room and all of its blurbs.
+  rpc DeleteRoom(DeleteRoomRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      delete: "/v1alpha3/{name=rooms/*}"
+    };
+    option (google.api.method_signature) = {
+      fields: ["name"]
+    };
+  }
+
+  // Lists all chat rooms.
+  rpc ListRooms(ListRoomsRequest) returns (ListRoomsResponse) {
+    option (google.api.http) = {
+      get: "/v1alpha3/rooms"
+    };
+  }
+
+  // Creates a blurb. If the parent field corresponds to a RoomBlurb
+  // resource, the blurb is understood to be a message in a chatroom. If the
+  // parent field is a ProfileBlurb resource name, the blurb is understood
+  // to be a post on a user profile.
+  rpc CreateBlurb(CreateBlurbRequest) returns (Blurb) {
+    option (google.api.http) = {
+      post: "/v1alpha3/{parent=rooms/*}/blurbs"
+      body: "*"
+      additional_bindings: {
+        post: "v1alpha3/{parent=users/*/profile}/blurbs"
+        body: "*"
+      }
+    };
+    option (google.api.method_signature) = {
```

Comment:
Optional: I think it would be preferable to put the things that do not change in front of the things that do (e.g. `blurb.user` before `blurb.text`/`blurb.image`).

Reply:
This is good feedback that shouldn't be optional but rather standardized (unless backwards incompatible of course).

Reply:
But there are cases where it is backwards-incompatible, so sadly it can not be a hard-and-fast rule.

There are a couple of legitimate reasons, also -- pretend that one of the two second arguments was _always_ sent but often the third argument was omitted (even if it is common to the two signatures). Then the ordering you used would be appropriate.

TL;DR: The ordering was wrong here. It is not a _per sé_ error, however.

Reply:
Ack.